pkg/build/types/types.go

```diff
@@ -23,72 +23,125 @@ import (
+	// The mutation's desired group ID
+	GID uint32
+	// The permission bits for the path
 	Permissions uint32
-	Source      string
-	Recursive   bool
+	// The source path to mutate
+	Source string
+	// Toggle whether to mutate recursively
+	Recursive bool
 }
 
 type OSRelease struct {
-	Name         string
-	ID           string
-	VersionID    string `yaml:"version-id"`
-	PrettyName   string `yaml:"pretty-name"`
-	HomeURL      string `yaml:"home-url"`
+	// Optional: The name of the OS
+	Name string
+	// Optional: The unique identifier for the OS
+	ID string
+	// Optional: The unique identifier for the version of the OS
+	VersionID string `yaml:"version-id"`
+	// Optional: The human readable description of the OS
+	PrettyName string `yaml:"pretty-name"`
+	// Optional: The URL of the homepage for the OS
+	HomeURL string `yaml:"home-url"`
+	// Optional: The URL of the bug reporting website for the OS
 	BugReportURL string `yaml:"bug-report-url"`
 }
 
 type ImageContents struct {
+	// A list of apk repositories to use for pulling packages
 	Repositories []string `yaml:"repositories,omitempty"`
-	Keyring      []string `yaml:"keyring,omitempty"`
-	Packages     []string `yaml:"packages,omitempty"`
+	// A list of public keys used to verify the desired repositories
+	Keyring []string `yaml:"keyring,omitempty"`
+	// A list of packages to include in the image
+	Packages []string `yaml:"packages,omitempty"`
 }
 
 type ImageEntrypoint struct {
-	Type          string
-	Command       string
+	// Optional: The type of entrypoint. Only "service-bundle" is supported.
+	Type string
+	// Required: The command of the entrypoint
+	Command string
+	// Optional: The shell fragment of the entrypoint command
 	ShellFragment string `yaml:"shell-fragment"`
 
 	// TBD: presently a map of service names and the command to run
 	Services map[interface{}]interface{}
 }
 
 type ImageAccounts struct {
-	RunAs  string `yaml:"run-as"`
-	Users  []User
+	// Required: The user to run the container as. This can be a username or UID.
+	RunAs string `yaml:"run-as"`
+	// Required: List of users to populate the image with
+	Users []User
+	// Required: List of groups to populate the image with
 	Groups []Group
 }
 
 type ImageConfiguration struct {
-	Contents    ImageContents     `yaml:"contents,omitempty"`
-	Entrypoint  ImageEntrypoint   `yaml:"entrypoint,omitempty"`
-	Cmd         string            `yaml:"cmd,omitempty"`
-	StopSignal  string            `yaml:"stop-signal,omitempty"`
-	WorkDir     string            `yaml:"work-dir,omitempty"`
-	Accounts    ImageAccounts     `yaml:"accounts,omitempty"`
-	Archs       []Architecture    `yaml:"archs,omitempty"`
+	// Required: The apk packages in the container image
+	Contents ImageContents `yaml:"contents,omitempty"`
+	// Required: The entrypoint of the container image
+	//
+	// This typically is the path to the executable to run. Since many of
+	// Chainguard Images do not include a shell, this should be the full path
+	// to the executable.
```

Comment:
I'd omit the chainguard piece to separate the OSS from product
```suggestion
	// This typically is the path to the executable to run. Since many of
	// images do not include a shell, this should be the full path
	// to the executable.
```